examples/hackweek_demo/plot_pnl.py

```diff
@@ -112,93 +113,134 @@ def get_wallet_from_onchain_trade_info(
     return wallet
 
 
-def calculate_pnl(trade_data, agent_list):
-    # pylint: disable=too-many-locals
-    """Calculates the pnl given trade data"""
-    # Drop all rows with nan maturity timestamps
-    trade_data = trade_data[~trade_data["maturity_time"].isna()]
-
-    # Filter trade_data based on agent_list
-    trade_data = trade_data[trade_data["operator"].isin(agent_list)]
-
-    if len(trade_data) == 0:
-        return (pd.DataFrame([]), pd.DataFrame([]))
-
-    # %% estimate position duration and add it in
-    position_duration = max(trade_data.maturity_time - trade_data.block_timestamp)
-    # print(f"empirically observed position_duration in seconds: {position_duration}")
-    # print(f"empirically observed position_duration in minutes: {position_duration/60}")
-    # print(f"empirically observed position_duration in hours: {position_duration/60/60}")
-    # print(f"empirically observed position_duration in days: {position_duration/60/60/24}")
-    position_duration_days = round(position_duration / 60 / 60 / 24)
-    # print(f"assuming position_duration is {position_duration_days}")
-    position_duration = position_duration_days * 60 * 60 * 24
-
-    agents = trade_data["operator"].value_counts().index.tolist()
-    agent_wallets = {
-        a: Wallet(
-            address=agents.index(a),
-            balance=types.Quantity(amount=FixedPoint(0), unit=types.TokenType.BASE),
-        )
-        for a in agents
-    }
-
-    # %%
-    pnl_data: list[dict[str, float]] = []
-    time_data: list[datetime] = []  # TODO: Figure out type of timestamps. datetime?
-    for _, row in trade_data.iterrows():
-        new_pnl: dict[str, float] = {}
-        for agent in agents:
-            agent_shortname = agent[:6] + "..." + agent[-4:]
-
-            marginal_trades = pd.DataFrame(row).T
-
-            # pass in one trade at a time, and get out the updated wallet
-            agent_wallets[agent] = get_wallet_from_onchain_trade_info(
-                address=agent,
-                trades=marginal_trades,
-                index=agents.index(agent),
-                add_to_existing_wallet=agent_wallets[agent],
+Info = NamedTuple(
```

Comment:
nice, but can we call it something specific like `AgentPlottingInfo`?  It looks like this is specific to plotting, not a generic `AgentInfo`?

Reply:
class is in with `ap = AgentPosition()`